token-lending/program/tests/withdraw_obligation_collateral.rs

```diff
@@ -0,0 +1,183 @@
+#![cfg(feature = "test-bpf")]
+
+mod helpers;
+
+use helpers::*;
+use solana_program_test::*;
+use solana_sdk::{
+    pubkey::Pubkey,
+    signature::{Keypair, Signer},
+    system_instruction::create_account,
+    transaction::Transaction,
+};
+use spl_token::instruction::approve;
+use spl_token_lending::{
+    instruction::withdraw_obligation_collateral,
+    math::Decimal,
+    processor::process_instruction,
+    state::{INITIAL_COLLATERAL_RATIO, SLOTS_PER_YEAR},
+};
+
+const LAMPORTS_TO_SOL: u64 = 1_000_000_000;
+const FRACTIONAL_TO_USDC: u64 = 1_000_000;
+
+#[tokio::test]
+async fn test_success() {
```

Comment:
This looks great! Could you add a failing test for this too?

Reply:
Done in 0e8810fc9feb30a819cf03675d90a69b78961db5. Also made the passing test more strict by withdrawing the maximum allowed amount.

Reply:
What's the right convention for tests like these that repeat a bunch of setup code? I get the itch to abstract

Reply:
Reducing setup code sounds great! The more setup, the more tedious it is to add more tests, which is definitely not great